packages/plugin/src/core-plugin.ts

```diff
@@ -55,6 +56,20 @@ export class CorePlugin<
       this.player,
       this.player.Event.MEDIA_LOADED,
       () => {
+        if (this._loadMediaEntryIdWorkaround) {
+          if (
+            this._loadMediaEntryIdWorkaround !== this.player.config.sources.id
+          ) {
+            throw new Error(
+              `Current contrib infrastructure does not allow load media of different entries. Expected entry id ${this._loadMediaEntryIdWorkaround}`
+            );
+          }
+
+          return;
+        }
+
+        this._loadMediaEntryIdWorkaround = this.player.config.sources.id;
```

Comment:
@esakal if i understood correctly - this will be called only on first load. If i'm right - please add a comment so the workaround logic will be clearer... If i'm wrong, please add a comment so it will be clear :)
I would also explain the workaround flow in a comment for the short memory developers. (also the reset / destroy process of it)